File: csc226/Erase.c
# include <stdio.h>
# include <string.h>
# include <stdlib.h>

int main(int argc, char const *argv[]) {
  int n;
  scanf("%d",&n);
  char *first = (char *) malloc(sizeof(char)*1000);
  char *second = (char *) malloc(sizeof(char)*1000);
  scanf("%s",first);
  scanf("%s",second);
  if (!n % 2) {
    if (!strcmp(first,second)) {
      printf("Deletion succeeded");
    } else {
      printf("Deletion failed");
    }
  } else {
    for (int i = 0; i < strlen(first); i++) {
      if (first[i] == second[i]) {
        printf("Deletion failed");
        return 0;
      }
    }
    printf("Deletion succeeded");
  }
  free(first);
  free(second);
  return 0;
}

File: csc226/Erase.py
#!/usr/bin/env python3
from sys import stdin

n = int(input())
(first,second) = [i.strip() for i in stdin.readlines()]
if (n % 2 == 0):
    if (first == second):
        print("Deletion succeeded")
    else:
        print("Deletion failed")
else:
    for i in range(len(first)):
        if (first[i] == second[i]):
            print("Deletion failed")
            exit()
    print("Deletion succeeded")
